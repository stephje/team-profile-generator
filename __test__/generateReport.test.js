const generateReport = require('../utils/generateReport.js');

const employeeDetails = {
  name: 'Stephanie Jenkins',
  id: 8,
  email: 's.jenkins3018@gmail.com',
  office: 8
};

describe("generateReport", () => {
    it("generates an empty report when there are no employees", () => {
      expect(generateReport([])).toMatchSnapshot();
    });

    it("generates a report containing a manger when given a manager", () => {
      expect(generateReport([employeeDetails])).toMatchSnapshot();
    });
  
  });