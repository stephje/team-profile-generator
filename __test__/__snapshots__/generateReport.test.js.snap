// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateReport generates a report containing a manger when given a manager 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <title>Team Profile</title>
    <!-- Google Fonts -->
    <link rel=\\"preconnect\\" href=\\"https://fonts.gstatic.com\\" />
    <link
      href=\\"https://fonts.googleapis.com/css2?family=Montserrat:wght@300;400&display=swap\\"
      rel=\\"stylesheet\\"
    />
    <link
      rel=\\"stylesheet\\"
      href=\\"https://use.fontawesome.com/releases/v5.8.1/css/all.css\\"
      integrity=\\"sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf\\"
      crossorigin=\\"anonymous\\"
    />
    <link rel=\\"stylesheet\\" href=\\"./style.css\\" />
  </head>

  <body>
    <header>
      <h1>Team Profile</h1>
    </header>

    <main>
      <div class=\\"managers flex\\"></div>
      <div class=\\"interns flex\\"></div>
      <div class=\\"engineers flex\\"></div>
    </main>
  </body>
</html>
"
`;

exports[`generateReport generates an empty report when there are no employees 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <title>Team Profile</title>
    <!-- Google Fonts -->
    <link rel=\\"preconnect\\" href=\\"https://fonts.gstatic.com\\" />
    <link
      href=\\"https://fonts.googleapis.com/css2?family=Montserrat:wght@300;400&display=swap\\"
      rel=\\"stylesheet\\"
    />
    <link
      rel=\\"stylesheet\\"
      href=\\"https://use.fontawesome.com/releases/v5.8.1/css/all.css\\"
      integrity=\\"sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf\\"
      crossorigin=\\"anonymous\\"
    />
    <link rel=\\"stylesheet\\" href=\\"./style.css\\" />
  </head>

  <body>
    <header>
      <h1>Team Profile</h1>
    </header>

    <main>
      <div class=\\"managers flex\\"></div>
      <div class=\\"interns flex\\"></div>
      <div class=\\"engineers flex\\"></div>
    </main>
  </body>
</html>
"
`;
